engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/SnapshotObject.java

```diff
@@ -191,7 +191,8 @@ public void processEvent(ObjectInDataStoreStateMachine.Event event) {
             s_logger.debug("Failed to update state:" + e.toString());
             throw new CloudRuntimeException("Failed to update state: " + e.toString());
         } finally {
-            if (event == ObjectInDataStoreStateMachine.Event.OperationFailed) {
+            DataObjectInStore obj = objectInStoreMgr.findObject(this, this.getDataStore());
+            if (event == ObjectInDataStoreStateMachine.Event.OperationFailed && !obj.getState().equals(ObjectInDataStoreStateMachine.State.Destroying)) {
```

Comment:
@koushik-das If there is failure in destroying snapshot then we should not delete db entry so that it can be used to detect which snapshots failed to delete successfully.